internal/pkg/term/progress/cloudformation_test.go

```diff
@@ -108,6 +108,44 @@ func TestRegularResourceComponent_Listen(t *testing.T) {
 		require.True(t, hasStarted, "the stopwatch should have started when an event was received")
 		require.Equal(t, time.Duration(0), elapsed)
 	})
+	t.Run("should keep timer running if multiple in progress events are received", func(t *testing.T) {
+		// GIVEN
+		ch := make(chan stream.StackEvent)
+		done := make(chan bool)
+		comp := &regularResourceComponent{
+			logicalID: "EnvironmentManagerRole",
+			statuses:  []stackStatus{notStartedStackStatus},
+			stopWatch: &stopWatch{
+				clock: &fakeClock{
+					wantedValues: []time.Time{testDate, testDate.Add(10 * time.Second)},
+				},
+			},
+			stream: ch,
+		}
+
+		// WHEN
+		go func() {
+			comp.Listen()
+			done <- true
+		}()
+		go func() {
+			ch <- stream.StackEvent{
+				LogicalResourceID: "EnvironmentManagerRole",
+				ResourceStatus:    "CREATE_IN_PROGRESS",
+			}
+			ch <- stream.StackEvent{
+				LogicalResourceID: "EnvironmentManagerRole",
+				ResourceStatus:    "CREATE_IN_PROGRESS",
+			}
+			close(ch) // Close to notify that no more events will be sent.
+		}()
+
+		// THEN
+		<-done // Wait for listen to exit.
+		elapsed, hasStarted := comp.stopWatch.elapsed()
+		require.True(t, hasStarted, "the stopwatch should have started when an event was received")
+		require.Equal(t, 10*time.Second, elapsed)
```

Comment:
Is there a way we can check that the status also hasn't changed? Or is that out of scope for this test?

Reply:
The status would change, as in we would add "create in progress" twice :P it's not really a big deal tho because we just display the last status while rendering the resource.

I changed the test slightly so that we check how many times `clock.now()` is called that validates this logic better. I had a bug in my first revision that @iamhopaul123 caught, this newer test should catch it